agent/src/main/kotlin/org/jetbrains/teamcity/vault/agent/VaultBuildFeature.kt

```diff
@@ -83,8 +83,15 @@ class VaultBuildFeature(dispatcher: EventDispatcher<AgentLifeCycleListener>,
                 try {
                     val template = createRestTemplate(settings, trustStoreProvider)
                     val authentication: ClientAuthentication = when (settings.auth.method) {
-                        AuthMethod.APPROLE -> {
-                            val wrapped = (settings.auth as Auth.AppRoleAuthAgent).wrappedToken
+                        AuthMethod.APPROLE,
+                        AuthMethod.LDAP,
+                        AuthMethod.USERPASS -> {
+                            val wrapped = when (val auth = settings.auth) {
+                                is Auth.AppRoleAuthAgent -> auth.wrappedToken
+                                is Auth.LdapAgent -> auth.wrappedToken
+                                is Auth.UserPassAgent -> auth.wrappedToken
+                                else -> throw IllegalArgumentException("unreachable")
```

Comment:
Maybe it's worth to note `settings.auth.method` and `settings.auth::class.name` in message, just in case. Also `fail/error` could be used.

Reply:
fixed